### ISSUE TITLE
Components: cookies are not passed through to edge lambda

### ISSUE THREAD

**Issue by josephluck** [danielcondemarin/serverless-next.js#154]
It looks like cookies from the request are not passed through from CloudFront to the underlying edge lambda. After deploying a next component, the CloudFront behaviour for the edge lambda looks like this:

![image](https://user-images.githubusercontent.com/11628330/65387297-ecf91080-dd3d-11e9-9b66-a0f84d3da1e8.png)

This is a problem for apps that rely on cookies for things like authentication etc, as the cookie header is not available in `getInitialProps`.

Perhaps it would be good to have some configuration options to specify the CloudFront behaviour , though it seems like a sensible default for next apps to have this configuration turned on.

It looks as though the behaviour for cookie forwarding is not configurable by the underlying [`@serverless/aws-cloudfront` component](https://github.com/serverless-components/aws-cloudfront/blob/master/lib/getDefaultCacheBehavior.js#L11). 

There's a [PR open](https://github.com/serverless-components/aws-cloudfront/pull/8) that will enable this configuration in the underlying cloudfront component, that can be utilised by this component. If it's merged and deployed, this issue can probably be closed as the default behaviour for forwarding cookies has been changed to forward by default.

**Comment by josephluck** [comment]
The PR that adds support for cookie forwarding has been merged, so I'll check whether this just works out-of-the-box or whether any additional changes are required soon 👍

**Comment by danielcondemarin** [comment]
> The PR that adds support for cookie forwarding has been merged, so I'll check whether this just works out-of-the-box or whether any additional changes are required soon 👍

From looking at this line https://github.com/serverless-components/aws-cloudfront/blob/master/lib/cacheBahaviorUtils.js#L2 it seems like we will need additional changes in the component to pass in `cookies: "all"` to the cloudfront forward config.

**Comment by danielcondemarin** [comment]
@josephluck This is now published on @1.2.3 🎉

**Comment by josephluck** [comment]
> > The PR that adds support for cookie forwarding has been merged, so I'll check whether this just works out-of-the-box or whether any additional changes are required soon 👍
> 
> From looking at this line https://github.com/serverless-components/aws-cloudfront/blob/master/lib/cacheBahaviorUtils.js#L2 it seems like we will need additional changes in the component to pass in `cookies: "all"` to the cloudfront forward config.

Turns out that https://github.com/serverless-components/aws-cloudfront/blob/master/lib/cacheBahaviorUtils.js#L2 is not actually used, since the value is always overridden by https://github.com/serverless-components/aws-cloudfront/blob/master/lib/getCacheBehavior.js#L15

So it looks like cookie forwarding is the default 🤷‍♂ . I must admit, the code is fairly tricky to parse...

**Comment by danielcondemarin** [comment]
@josephluck Good spot. In any case it’s better to be explicit imo, so in the unlikely event the cloudfront component defaults change in future, the component is not affected

**Comment by josephluck** [comment]
> @josephluck Good spot. In any case it’s better to be explicit imo, so in the unlikely event the cloudfront component defaults change in future, the component is not affected

Totally agreed.

I tried it out last night and it looks like the query string and cookies are forwarded for the CloudFront behaviours attached to the S3 buckets correctly, but _not_ for the edge lambda. Strange as from the CloudFront SDK docs, it looks to me that everything is set up correctly (though not sure if there is separate cache configuration for edge lambdas specifically).

Hopefully I'll have time over the next few days to take a look, but I found it quite difficult to map the configuration through the components when I tried initially.
